config_role.go

```diff
@@ -0,0 +1,96 @@
+package evergreen
+
+import (
+	"time"
+
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// LDAPRoleMap contains a mapping of LDAP groups for a user to roles. LDAP
+// groups are represented by their name and roles by their unique ID in the
+// roles collection.
+type LDAPRoleMap struct {
+	LastUpdated time.Time         `bson:"last_updated" json:"last_updated" yaml:"last_updated"`
+	Map         map[string]string `bson:"map" json:"map" yaml:"map"`
```

Comment:
I think this might be more future-proof if it were a list of structs that contain both the LDAP group and role. Someone could potentially create a group that has an illegal character in it, and we would be able to persist it but will have trouble querying it. Having a struct also makes it clearer whether the keys are groups or roles